crates/nu-cmd-dataframe/src/dataframe/values/nu_dataframe/conversion.rs

```diff
@@ -963,10 +916,154 @@ fn series_to_values(
+        AnyValue::UInt16(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::UInt32(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::UInt64(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::Int8(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::Int16(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::Int32(i) => Ok(Value::int(*i as i64, span)),
+        AnyValue::Int64(i) => Ok(Value::int(*i, span)),
+        AnyValue::Float32(f) => Ok(Value::float(*f as f64, span)),
+        AnyValue::Float64(f) => Ok(Value::float(*f, span)),
+        AnyValue::Date(d) => {
+            let nanos = nanos_per_day(*d);
+            datetime_from_epoch_nanos(nanos, &None, span)
+                .map(|datetime| Value::date(datetime, span))
+        }
+        AnyValue::Datetime(a, time_unit, tz) => {
+            let nanos = nanos_from_timeunit(*a, *time_unit);
+            datetime_from_epoch_nanos(nanos, tz, span).map(|datetime| Value::date(datetime, span))
+        }
+        AnyValue::Duration(a, time_unit) => {
+            let nanos = match time_unit {
+                TimeUnit::Nanoseconds => *a,
+                TimeUnit::Microseconds => *a * 1_000,
+                TimeUnit::Milliseconds => *a * 1_000_000,
+            };
+            Ok(Value::duration(nanos, span))
+        }
+        // AnyValue::Time represents the current time since midnight.
+        // Unfortunately, there is no timezone related information.
+        // Given this, calculate the current date from UTC and add the time.
+        AnyValue::Time(nanos) => time_from_midnight(*nanos, span),
+        AnyValue::List(series) => {
+            series_to_values(series, None, None, span).map(|values| Value::list(values, span))
+        }
+        AnyValue::Struct(_idx, _struct_array, _s_fields) => {
+            // This should convert to a StructOwned object.
+            let static_value =
+                any_value
+                    .clone()
+                    .into_static()
+                    .map_err(|e| ShellError::GenericError {
+                        error: "Cannot convert polars struct to static value".into(),
+                        msg: e.to_string(),
+                        span: Some(span),
+                        help: None,
+                        inner: Vec::new(),
+                    })?;
+            any_value_to_value(&static_value, span)
+        }
+        AnyValue::StructOwned(struct_tuple) => {
+            let values: Result<Vec<Value>, ShellError> = struct_tuple
+                .0
+                .iter()
+                .map(|s| any_value_to_value(s, span))
+                .collect();
+            let fields = struct_tuple
+                .1
+                .iter()
+                .map(|f| f.name().to_string())
+                .collect();
+            Ok(Value::Record {
+                val: Record {
+                    cols: fields,
+                    vals: values?,
+                },
```

Comment:
In the future we will make those fields private and force you to go through either `.collect` or `Record::from_raw_cols_vals` or similar helper. 

But I can take care of updating that in a separate PR, already started a branch for those remaining uses when I came across the unsafe issue.